### PR TITLE
Throw error on unknown sections in markdown format

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Add test to ensure that empty sections in markdown format are treated correctly.
- Parse unknown sections in markdown format in order to throw an error.

fixes https://github.com/RasaHQ/rasa/issues/4304

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: data/test/markdown_single_sections/not_existing_section.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## synonym:animal
```
Comment (wochinge) [inline data/test/markdown_single_sections/not_existing_section.md]: what's the difference to the `empty_section` one?

File: rasa/nlu/training_data/formats/markdown.py [hunk 2]
```diff
@@ -37,10 +37,8 @@
 ESCAPE = re.compile(r"[\b\f\n\r\t]")
 
 
-def encode_string(s):
-    """Return a encoded python string
-
-    """
+def encode_string(s: Text) -> Text:
+    """Return a encoded python string"""
```
Comment (wochinge) [inline rasa/nlu/training_data/formats/markdown.py]:
```suggestion
    """Return a encoded python string."""
```

File: rasa/nlu/training_data/formats/markdown.py [hunk 3]
```diff
@@ -55,13 +53,13 @@ class MarkdownReader(TrainingDataReader):
     """Reads markdown training data and creates a TrainingData object."""
 
     def __init__(self) -> None:
-        self.current_title = None
-        self.current_section = None
-        self.training_examples = []
-        self.entity_synonyms = {}
-        self.regex_features = []
+        self.current_title: Optional[Text] = None
```
Comment (wochinge) [inline rasa/nlu/training_data/formats/markdown.py]: have we already dropped Python 3.5 supports? Type annotation for variables are not supported in Py 3.5

File: rasa/nlu/training_data/formats/markdown.py [hunk 4]
```diff
@@ -117,7 +121,7 @@ def _load_files(self, line):
                     {"name": self.current_title, "elements": str(fname)}
                 )
 
-    def _parse_item(self, line):
+    def _parse_item(self, line: Text) -> None:
```
Comment (wochinge) [inline rasa/nlu/training_data/formats/markdown.py]: great work in adding all the type annotations!

File: rasa/nlu/training_data/formats/markdown.py [hunk 5]
```diff
@@ -90,22 +88,28 @@ def _strip_comments(text: Text) -> Text:
...
+        def make_regex(section_name: Text) -> Pattern:
             return re.compile(r"##\s*{}:(.+)".format(section_name))
 
         return {sn: make_regex(sn) for sn in section_names}
 
-    def _find_section_header(self, line):
+    def _find_section_header(self, line: Text) -> Optional[Tuple[Text, Text]]:
         """Checks if the current line contains a section header
         and returns the section and the title."""
         for name, regex in self.section_regexes.items():
```
Comment (wochinge) [inline rasa/nlu/training_data/formats/markdown.py]: do we even need this if we have 

```
match = re.search(r"##\s*(.+):(.+)", line)
if match is not None:
     return match.group(1), match.group(2)
```

?

File: data/test/markdown_single_sections/not_existing_section.md (hunk 1 above)
Comment (tabergma) [reply data/test/markdown_single_sections/not_existing_section.md]: this file contains a not existing section, e.g. `entity`. the issue mentioned that we just ignored unknown sections before and we just added all values of that unknown section to the section that was listed before.

File: rasa/nlu/training_data/formats/markdown.py (hunk 3 above)
Comment (tabergma) [reply rasa/nlu/training_data/formats/markdown.py]: good point, forgot about it, will remove them again.

File: rasa/nlu/training_data/formats/markdown.py (hunk 5 above)
Comment (tabergma) [reply rasa/nlu/training_data/formats/markdown.py]: true, will change it.

File: data/test/markdown_single_sections/not_existing_section.md (hunk 1 above)
Comment (wochinge) [reply data/test/markdown_single_sections/not_existing_section.md]: ah, `entity` looked to valid to me 🙈
